### DISCUSSION TITLE
USTC 编译原理与技术 - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#672] [Announcements]
# USTC 编译原理与技术 - CS自学指南

CS自学指南

https://csdiy.wiki/%E7%BC%96%E8%AF%91%E5%8E%9F%E7%90%86/USTC-Compilers/

**Comment by XZQshiyu** [comment]
顺手贴个24年课程主页https://ustc-compiler-2024.github.io/homepage/

**Reply by w666cy** [reply]
抓到一只助教🤗

**Reply by zzXavier** [reply]
这是哪个学校的呀

**Comment by w666cy** [comment]
中国科学技术大学
University of Science and Technology of China

2024-10-23 18:50:32 "zzXavier" ***@***.***> 写道：

这是哪个学校的呀

**Comment by zerorobot0** [comment]
诶？之前的评论呢？里边大家总是会补充或分享一些资源，怎么不见啦🥹

**Comment by MingZ-Jim** [comment]
好像在前言更新的时候大家的评论都会重置

**Comment by ColdCodeCool** [comment]
很棒，很厉害。但有个疑问，就是北大的计算机课程不够solid吗？我意思是，正常情况下你们应该应付自己的课程project和ddl都挺困难了，还有这么多时间自学吗？

**Comment by YiXun-13468878779** [comment]
有交流群吗？想求一个

**Reply by hanquanjushi** [reply]
之前cs143底下有一个904654497

**Comment by JamesBrianD** [comment]
请问有没有LLM相关的公开课推荐

**Reply by hanquanjushi** [reply]
infrastructure可以看from zero to hero，入门看李宏毅近一两年的课

**Reply by hanquanjushi** [reply]
USTC的AI交流群985029820

**Comment by sqlvlq123** [comment]
我从b 站辗转到你这个网站。。。

**Comment by Thea1788222** [comment]
很想知道作者在三年里学习这些课程的时间分布，想在总体上对照了解一下自己的目前所处的位置和进度，谢谢

**Reply by hanquanjushi** [reply]
My reference value is not great, because like some USTC students, I spent a lot of time studying mathematical & physics courses in the first two years of university, and I have little time to learn CS61A, CSAPP and other open courses. In fact, I think such excellent open courses need to spend more than half a year to write experiments to learn well, which is  difficult for computer students with heavy study. In our school, there are also outstanding students in the first and second years of the study of courses, go to the industry internship research AI, I think this is a more ideal choice. In addition, enter a university laboratory during the first two years of undergraduate study  is also a good choice, under the leadership of doctoral students, the improvement of scientific research and tool use ability is often very fast.

**Comment by angl1n** [comment]
cs61a要账号密码了咋办

**Reply by Chebetk774** [reply]
同问一个，有点抓心挠肝了

**Reply by t3qvi1a** [reply]
这个网站可以，20fall的 https://web.archive.org/web/20210116184445/https://cs61a.org/

**Comment by SuperGODOG** [comment]
还算稳定的机场https://in.mesl.cloud/#/register?code=KxtStC3L

**Comment by ourcx** [comment]
你好

**Comment by xiaocup** [comment]
建了个Q群，有需要可以加入  ：765696882  验证答案 CS自学指南

**Comment by happyeyeryday** [comment]
真不错

**Comment by fanan226688** [comment]
如果不是本校学生，中国科技大学的编译原理视频课是不是看不了?

**Reply by fanan226688** [reply]
b站上的是实验课内容

**Reply by Terialion** [reply]
https://ustc-compiler-2025.github.io/homepage/ 可以来课程主页看ppt，我们每节课都会同步slide，目前录课只有本校学生能看

**Reply by fanan226688** [reply]
谢谢！

**Comment by Terialion** [comment]
25秋的课程主页 https://ustc-compiler-2025.github.io/homepage/

**Reply by Terialion** [reply]
对实验有什么疑问以及改进意见欢迎联系~

**Reply by gulugulubing** [reply]
感谢，内容很丰富。就是试着做了下lab4，刚发现提供龙芯的工具只有x86的

**Reply by gulugulubing** [reply]
licm的框架代码中，判断要不要插入preheader的条件似乎有误，要自己改一下~

**Reply by AkunCube** [reply]
23版本的后面lab5和lab6框架代码有点太随意了，像是好几个人写的
